### PR TITLE
Feature: Add users and user detail routes

### PR DESCRIPTION
**Pull request by samanehsan**
## Ticket
https://openscience.atlassian.net/browse/EOSF-14

# Purpose
Add users and user detail routes

# Notes for Reviewers
The `/users` route includes both the current user's record and a duplicate "me" record.

### Routes Added/Updated
- `/users`
- `/users/user_id`

### CODE REVIEW

File: addon/models/user.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import DS from 'ember-data';
 import OsfModel from 'ember-osf/models/base';
 
 export default OsfModel.extend({
+    full_name: DS.attr('string'),
```
Comment (samchrisinger) [inline addon/models/user.js]: These fields should really be camelCased (my fault)-- can you please change them?

**Comment by samanehsan** [comment]
Done making that change @samchrisinger!

File: tests/dummy/app/routes/users/detail.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import Ember from 'ember';
+
+export default Ember.Route.extend({
```
Comment (samchrisinger) [inline tests/dummy/app/routes/users/detail.js]: Since the parent route uses the AuthenitcatedRouteMixin, this one probably should too.

Comment (samchrisinger) [inline tests/dummy/app/routes/users/detail.js]: see: http://ember-simple-auth.com/api/classes/AuthenticatedRouteMixin.html

**Comment by samchrisinger** [comment]
@samanehsan one small change, otherwise lgtm

**Comment by samanehsan** [comment]
Made that change @samchrisinger! 🚀

**Comment by samchrisinger** [comment]
Merged manually. Thanks!

**Pull request closed without merging**